src/contrib/writing-design-page.rst

```diff
@@ -1,2 +1,127 @@
-How to write a design page
-##########################
+Writing a Design Page
+#####################
+
+A design page is a document that describes how a new feature is going to be
+implemented. However, the purpose of the design is not to document the feature
+itself (we have other channels for that) but rather to document the decision
+process behind the final solution. That means, if multiple solutions were
+considered, the design page should describe why the final solution was selected
+and not the others.
+
+.. note::
+
+    **Writing a design page on a complex features or changes is very beneficial
+    to you, the author of these changes.** SSSD, identity management and Linux
+    is a huge world and it is not possible to know, anticipate and understand
+    everything. Sharing a design page with multiple people before writing the
+    code can give you a valuable feedback that helps you to avoid errors and
+    reinventing the wheel since the reviewers can point you to an existing API
+    and catch corner cases that you were not aware of.
+
+Before you start writing a design page, **think about the audience**. It is
+important to know the expected audience so you know how thorough you have to be
+in your explanations. For SSSD project, we can assume that the audience are
+SSSD developers and people that are more or less familiar with SSSD and other
+Identity Management products therefore you do not have to go in depth in these
+areas and you can mostly focus on the change itself.
+
+We usually start writing design page using Google Docs or similar online
+services since it gives the reviewers lots of useful tools such as comments,
+assignments, suggestions and revision history. But please, do not forget to
+submit a pull request to `sssd.io`_ to include the final revision `here`_ so it
```

Comment:
At the end of the document we provide `ReST` template. I think we should mention here that `ReST` is the preferred format. Also what will happen when 3rd party developer will provide design document in for example PDF format? If it will be still accepted?

Reply:
Done.